src/database/migrations/20201118202558-user-uuid.ts

```diff
@@ -0,0 +1,53 @@
+import { QueryInterface, DataTypes } from 'sequelize';
+
+export default {
+    up: async (queryInterface: QueryInterface): Promise<void> => {
+        // Transactions are automatically use because a namespace is injected into sequelize when fetching configurations
+        await queryInterface.sequelize.transaction(async () => {
+            await queryInterface.addColumn('users', 'user_uuid', {
+                type: DataTypes.UUID,
+                allowNull: true,
+                unique: true,
+                defaultValue: DataTypes.UUIDV4
+            });
+
+            await queryInterface.sequelize.query(
+                'UPDATE users SET user_uuid = uuid_generate_v4()'
+            );
+
+            await queryInterface.changeColumn(
+                'users',
+                'user_uuid',
+                {
+                    type: DataTypes.UUID,
+                    allowNull: false,
+                    unique: true,
+                    defaultValue: DataTypes.UUIDV4
```

Comment:
Is this a value or a type?

Reply:
This is an enum that represents a function that returns a value: like DataTypes.NOW

UUID is the type but it can be a v1 or a v4